core/src/main/scala/cats/Invariant.scala

```diff
@@ -128,7 +128,7 @@ object Invariant extends ScalaVersionSpecificInvariantInstances with InvariantIn
     cats.catsInstancesForId
   @deprecated("Added for bincompat", "2.8.0")
   @cats.compat.targetName("catsInstancesForId")
```

Comment:
Could you provide some details on how this guy is supposed to work please? I think it might help me with my work in #4187.

Reply:
Definitely, do you mean about the compat thing or the semantics of `targetName` which is a Scala 3 feature? You can see how it works in https://github.com/typelevel/cats/issues/4143#issuecomment-1085761689.